agent/command/git.go

```diff
@@ -208,6 +215,9 @@ func (opts cloneOpts) buildSSHCloneCommand() ([]string, error) {
 	if opts.branch != "" {
 		cloneCmd = fmt.Sprintf("%s --branch '%s'", cloneCmd, opts.branch)
 	}
+	if opts.cloneParams != "" {
+		cloneCmd = fmt.Sprintf("%s %s", cloneCmd, opts.cloneParams)
```

Comment:
If both branch and cloneParams are defined, do we want to combine both in the cloneCmd?

Reply:
yes! i think one of the use cases is --single-branch and branch which work together to only clone the specified branch

Reply:
But can a user specify both --branch and --cloneParams? Right now, it seems like we overwrite the cloneCmd if cloneParams are defined, but we don't seem to be blocking the user from specifying both. If overwriting it is the desired behavior, then maybe we should error if the user specifies both?

Reply:
ah the code is currently adding on cloneParams to the end of cloneCmd if that's what you're suggesting (unless I'm reading it wrong?)

Reply:
Nope, I was reading it wrong :)